### PR TITLE
Adjusted miner, added intervals, initial reward rework

### PR DESCRIPTION
**Pull request by reed987**
- Added miner properties file
- Check for  supported crypto symbols
- Added intervals
- Check for supported intervals

### CODE REVIEW

File: neurons/miner.py [hunk 1]
```diff
@@ -62,17 +62,18 @@ def __init__(self, config=None):
...
             model_path = config.get('DEFAULT', 'model_path', fallback=None)
-            currency = config.get('DEFAULT', 'currency', fallback='usd')  # Default to 'usd' if not specified
+            currency = config.get('DEFAULT', 'currency', fallback='usd')  # Default to 'usd'
+            intervals = config.get('DEFAULT', 'intervals', fallback='5m').split(',')  # Default to '5m'
             
-            return [symbol.strip() for symbol in symbols], model_path, currency  # Strip whitespace from symbols
+            return [symbol.strip() for symbol in symbols], model_path, currency, intervals  # Strip whitespace from symbols
         
-        return [], None, 'usd'  # Return empty list, None for model path, and default currency
+        return [], None, 'usd', '5m'  # Return empty list, None for model path, default currency and intervals
```
Comment (servant321) [inline neurons/miner.py:76]: is the 5m the time frame interval? should it be hardcoded?

File: neurons/miner.py [hunk 2]
```diff
@@ -167,16 +168,25 @@ def generate_prediction(self, historical_prices):
...
             bt.logging.warning(msg)
             return synapse
 
+        # check if provided intervals are supported by the miner
+        is_valid_interval = all(interval in self.intervals for interval in requested_intervals)
+        if not is_valid_interval:
+            msg = "No supported intervals provided"
+            synapse.dummy_output = msg
+            bt.logging.warning(msg)
+            return synapse
```
Comment (servant321) [inline neurons/miner.py:190]: we could add an error field in the synapse so that the validator server can also see the issue

Comment (XDarnoXx) [reply neurons/miner.py:188]: test

Comment (XDarnoXx) [reply neurons/miner.py:188]: test

File: neurons/miner.py (hunk 1 above)
Comment (reed987) [reply neurons/miner.py:76]: It is just a default value but the code will fail if this case happens as we cannot continue without a model.

File: neurons/miner.py (hunk 2 above)
Comment (reed987) [reply neurons/miner.py:190]: Done.

**Pull request merged**